Source/Urho3D/Math/Vector3.h

```diff
@@ -435,61 +277,63 @@ class URHO3D_API Vector3
     unsigned ToHash() const
     {
         unsigned hash = 37;
-        hash = 37 * hash + FloatToRawIntBits(x_);
-        hash = 37 * hash + FloatToRawIntBits(y_);
-        hash = 37 * hash + FloatToRawIntBits(z_);
-
+        hash = 37 + (x_ * hash);

```

Comment:
Why changed? Such changed could potentially break someone's caches or so on.

Reply:
Because it can be `BaseVector3<char>` or `BaseVector3<double>` or `BaseVector3<int64_t>`. Use of `FloatToRawIntBits()` becomes invalid in many contexts.

Reply:
However, the fix is unacceptable.
1. It's bad to change hashing algo, people may have assets that rely on it.
2. The new algo is bad itself. (0, 0, 0) and (0.01, 0, 0) will have the same hashes. (1kkkk, 0, 0) will just explode hashing algo and end up in UB.

Reply:
Hmm true. I will add one more template specialization restoring old algorithm for floats here. For other types it should be sufficient to pick large enough prime right? How about `16777619` and  `1099511628211` used by fnv?